tests/cupy_tests/logic_tests/test_truth.py

```diff
@@ -125,6 +125,40 @@ def test(self, xp, dtype):
         return getattr(xp, self.f)(x, y, self.assume_unique, self.invert)
 
 
+class TestIntersect1d:
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_one_dim_with_unique_values(self, xp, dtype):
+        a = xp.array([1, 2, 3, 4, 5], dtype=dtype)
+        b = xp.array([1, 2, 3, 4, 5], dtype=dtype)
+        return xp.intersect1d(a, b, assume_unique=True)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_with_random_val(self, xp, dtype):
+        a = xp.array([3, 4, 9, 1, 5, 4], dtype=dtype)
+        b = xp.array([8, 7, 3, 9, 0], dtype=dtype)
+        return xp.intersect1d(a, b)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_more_dim(self, xp, dtype):
+        a = testing.shaped_random((3, 4), xp, dtype=dtype)
+        b = testing.shaped_random((5, 2), xp, dtype=dtype)
+        return xp.intersect1d(a, b)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_return_indices(self, xp, dtype):
+        a = xp.array([2, 3, 4, 1, 9, 4], dtype=dtype)
+        b = xp.array([7, 5, 1, 2, 9, 3], dtype=dtype)
+        c, i1, i2 = xp.intersect1d(a, b, return_indices=True)
+        ui1 = xp.unravel_index(i1, a.shape)
+        ui2 = xp.unravel_index(i2, b.shape)
+        return c, a[ui1], b[ui2]
```

Comment:
According to [numPy.intersect1d documentation](https://numpy.org/doc/stable/reference/generated/numpy.intersect1d.html), the first instance of a value is used if there are multiple.
```suggestion
        return xp.intersect1d(a, b, return_indices=True)
```

Reply:
Could you add a new test case for this check?

Reply:
Ah, yes! I missed it. Thanks for pointing out :)

Reply:
Could you fix also this line? https://github.com/cupy/cupy/pull/6402/files#r795418783

Reply:
Yes, done :+1: